File: SourceCode/agent/src/com/netmessenger/agent/agentkaixin/datastore/IRecipientInfoDAO.scala
package com.netmessenger.agent.agentkaixin.datastore;

import java.util.List;


 trait IRecipientInfoDAO {

	def isExist(recipientInfo: RecipientInfo):Boolean;

	def add( recipientInfo:RecipientInfo):Unit;

	def countRecipients():Int;

	def goThroughAll(func:(RecipientInfo) => Unit):Int;
	
	def save():Unit;
	
	def close():Unit;

}

File: SourceCode/agent/src/com/netmessenger/agent/agentkaixin/datastore/IRecipientInfoDAO.java
package com.netmessenger.agent.agentkaixin.datastore;

import java.util.List;


public interface IRecipientInfoDAO {

	public abstract boolean isExist(RecipientInfo recipientInfo);

	public abstract void add(RecipientInfo recipientInfo);

	public abstract int countRecipients();

	public List<RecipientInfo> findAll();
	
	public abstract void save();
	
	public abstract void close();

}